docs/JAX_machine_translation.md

```diff
@@ -0,0 +1,502 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# NLP: JAX Machine Translation
+
+```{code-cell} ipython3
+## adapted from https://keras.io/examples/nlp/neural_machine_translation_with_transformer/, which is itself an adaptation from https://www.manning.com/books/deep-learning-with-python-second-edition
+```
+
+```{code-cell} ipython3
+import os
+
+import pathlib
+import random
+import string
+import re
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+import optax
+
+from flax import nnx
+
+import tiktoken
+```
```

Comment:
It may make sense to add a subsection with additional requirements description and installation: for example tiktoken and grain, tqdm